agent/pkg/singularity/singularity.go

```diff
@@ -359,7 +359,10 @@ func (s *SingularityClient) RunContainer(
 		fmt.Sprintf("SINGULARITYENV_CUDA_VISIBLE_DEVICES=%s", cudaVisibleDevicesVar),
 		fmt.Sprintf("APPTAINERENV_CUDA_VISIBLE_DEVICES=%s", cudaVisibleDevicesVar),
 	)
-
+	httpsProxy := os.Getenv("https_proxy")
```

Comment:
You need the triple `http_proxy, https_proxy, no_proxy`  for things to work properly.

Reply:
Testing has shown this to be sufficient for the case at hand. Do you need me to add the others?